text-utils/more.c

```diff
@@ -1076,13 +1076,18 @@ static void sigwinch_handler(struct more_control *ctl)
 	}
 }
 
+static void free_args(char ***args)
+{
+	free(*args);
+}
+
 static void execute(struct more_control *ctl, char *filename, char *cmd, ...)
 {
 	int id;
 	int n;
 	va_list argp;
 	char *arg;
-	char **args;
+	char **args __attribute__((__cleanup__(free_args))) = NULL;;
```

Comment:
This is the first place in util-linux we have automatic cleanup :-)

Not sure if this is the right way how to introduce it. Maybe it would be better to NOT do it in more(1) re-engineering, but create another PR and add some free()-like calls and initialization macros to xalloc.h? Maybe also discuss it on mailing list.

Reply:
Good point. While I think cleanup attribute should be taken in use this is not the best way to introduce it to project. I will drop the `avoid using alloca() function` commit from updated pull.

Secondly making a separate change request that will add cleanup attribute and related functions to xalloc.h is better way. That effort is obviously now backlogged, and delivery will happen with some delay. Personally I think moving to automatic cleanups in selected code paths would make code better.